var/spack/repos/builtin/packages/legion/package.py

```diff
@@ -148,42 +148,22 @@ def validate_gasnet_root(value):
         else:
             return True
 
-    variant(
-        "gasnet_root",
-        default="none",
-        values=validate_gasnet_root,
-        description="Path to a pre-installed version of GASNet (prefix directory).",
-        multi=False,
-    )
-    conflicts("gasnet_root", when="network=mpi")
-
-    gasnet_conduits = ["aries", "ibv", "udp", "mpi", "ucx", "ofi-slingshot11"]
-    variant(
-        "conduit",
-        default="none",
-        values=gasnet_conduits + ["none"],
-        description="The gasnet conduit(s) to enable.",
-        multi=False,
-    )
-
-    conflicts(
-        "conduit=none",
-        when="network=gasnet",
-        msg="a conduit must be selected when 'network=gasnet'",
-    )
-
-    for c in gasnet_conduits:
-        conflict_str = "conduit=%s" % c
-        conflicts(
-            conflict_str, when="network=mpi", msg="conduit attribute requires 'network=gasnet'."
+    with when("network=gasnet"):
+        variant(
+            "gasnet_root",
+            default="none",
+            values=validate_gasnet_root,
+            description="Path to a pre-installed version of GASNet (prefix directory).",
+            multi=False,
         )
-        conflicts(
-            conflict_str, when="network=none", msg="conduit attribute requires 'network=gasnet'."
+        variant(
+            "conduit",
+            default="mpi",
```

Comment:
Here I picked one randomly, let me know if anything else should be the default.

Reply:
`mpi` definitely shouldn't be the default.  It's there for debug rather than performance.  The right conduit is system dependent, and I am not aware of a good way to automatically detect which of the high-performance conduits (aries, ibv, ucx, ofi) is appropriate for a given system.  Pinging @bonachea for best practices here.

Reply:
The right conduit depends on a number of system-dependent factors, many of which are difficult to automatically detect, especially from Spack. Unfortunately the performance penalties for getting this wrong are quite high, and I agree with @streichler that "mpi" is a very bad default for this reason.

Other GASNet clients (e.g. UPC++) build for all available GASNet conduits and allow the end user to control conduit selection post-install, but I believe Realm lacks that capability.  I think the best practice for this package is to maintain the status quo where `default="none"` and force the installing user to select a conduit appropriate for their network hardware.

```suggestion
            default="none",
```

Reply:
Ok, so a "none" and a conflict that makes "none" not a valid value.

Reply:
I think the constraint is that when `conduit=none` then `network` must be something other than `gasnet` (i.e., `mpi`, `ucx` or `none`), and conversely if `network=gasnet` then `conduit` cannot be `none`. I.e., `conduit` is really only meaningful for the GASNet build and should be set iff GASNet is enabled.

Reply:
I suppose if there were a way for the `conduit` variant to exist *only* when `network=gasnet`, that would be appropriate. I don't know enough about Spack's package format to say if that's possible or reasonable.

Reply:
@elliottslaughter That is already the case. This PR makes the `conduit` variant conditional on `network=gasnet`. That means the variant doesn't even exist for other `network` values. 

What I am saying is that I can make the default something that will require a user to pick one of the other values. Is that ok?

Reply:
To be clear, currently if a user picks `network=gasnet` and doesn't explicitly give `conduit=*` clingo will pick one of the valid values randomly.

Reply:
> This PR makes the `conduit` variant conditional on `network=gasnet`. That means the variant doesn't even exist for other `network` values.

Ok, good.

> What I am saying is that I can make the default something that will require a user to pick one of the other values. Is that ok?

Yes, I think that's what we want.

Reply:
@elliottslaughter Ok, done